src/components/text/index.tsx

```diff
@@ -124,16 +125,16 @@ export function Text({
             setThing(dataset, updatedResource),
             { fetch }
           );
-
-          setDataset(savedDataset);
+          const latestDataset = await getSolidDataset(saveDatasetTo);
```

Comment:
One consideration could be to turn this into a small helper function. 
Perhaps something like `updateStateDataset` which would accept two parameters

- The `sourceUrl` which is in the form of `saveDatasetTo` or `getSourceUrl(dataset)`
- The `setState` function where the result will be set


It's just something to consider but it could help reduce some of the duplications